packages/gcc10.rb

```diff
@@ -35,19 +39,27 @@ class Gcc10 < Package
   depends_on 'isl'
   depends_on 'glibc'
   depends_on 'zstd'
-
+  
   def self.preinstall
-    gccver = `gcc -v 2>&1 | tail -1 | cut -d' ' -f3`.chomp
-    abort "GCC version #{gccver} already installed.".lightgreen unless "#{gccver}" == "No" || "#{gccver}" == "not" || "#{gccver}" == "gcc:" || "#{gccver}" == "#{version}"
+    installed_gccver = `gcc -v 2>&1 | tail -1 | cut -d' ' -f3`.chomp
+    gcc_version = version.split("-")[0]
+    abort "GCC version #{installed_gccver} already installed.".lightgreen unless "#{installed_gccver}" == "No" || "#{installed_gccver}" == "not" || "#{installed_gccver}" == "gcc:" || "#{installed_gccver}" == "#{gcc_version}"
```

Comment:
This section looks too hackish.  There has got to be a more elegant way to do this.

Reply:
Is your concern with the abort line or the generation of the installed_gccver value?

My only change here is to add logic to look at the version in the package file and chop off the dash-subversion since we aren't using a tagged build. (But this is based off of the 10.2.x stable branch.)

Reply:
I was the one who added this logic initially so I deserve all the blame.  I just hate everything about how this works.

Reply:
Ah. As my initial goal was just to get a version of gcc that would compile, I was trying to have a light touch on everything else. 

Honestly, I'd just get rid of the whole block.

I think it is better to handle ruby failing due to gcc being uninstalled by putting a wrapper on ruby, not by keeping people from messing up their gcc installs.

(If someone messes up their gcc install.... well... it's a learning experience... nobody is going to do it manually.)

But then you have probably dealt with enough people mucking it up to want to have a check here, since you wrote that code, yes?

Reply:
Well, the idea initially was to prevent someone from reinstalling gcc and bricking crew.  Since we can now switch gcc versions somewhat gracefully, it might be good to keep this logic.